### PR TITLE
v2 draft

### PR DESCRIPTION
**Pull request by Portals**
If you have read v1, read the migration first, please.

### CODE REVIEW

File: react/README.md [hunk 1]
```diff
@@ -25,35 +24,24 @@ If you're unsure about the tools used or ReactJS generally, then you shouldn't b
...
-
-### View
-
-A view can be a class component or a functional component with hooks. It may contain other views or [elements](#element), but not contain [screens](#screen) or [use-cases](#use-case).
-
-An example for what a view could be:
+### Use-case
 
--   A LoginForm: A view that has multiple textfields and saves the state of them.
+Your web application will contain several use-cases, who in their turn can contain different screens. A use-case can, for example, be a 'Users'. This use-case manages the creation, editing, deletion, and viewing of users. This is a neat way of structuring your project.
```
Comment (ericlp) [inline react/README.md]:
```suggestion
Your web application will contain several use-cases, who in their turn can contain different screens. A use-case can, for example, be 'Users'. This use-case manages the creation, editing, deletion, and viewing of users. This is a neat way of structuring your project.
```

File: react/README.md [hunk 2]
```diff
@@ -9,6 +9,277 @@ For migration between style guides, please see [MIGRATION.md](MIGRATION.md).
...
+
+---
+
+## Translations
+
+There are three ways of implementing translations in your app:
+
+1. Have one big `Translations.json` file with all translations.
+2. Have a `CommonTranslations.json` file common translations such "Cancel" or "Save", and one `*.json` file per `use-case`. 
+3. Still have a `CommonTranslation.json`, but have `*.json` everywhere it's needed.
```
Comment (ericlp) [inline react/README.md]: What is the difference?

Comment (Portals) [reply react/README.md]: Is this better?

```suggestion
2. Have a `CommonTranslations.json` file common translations such "Cancel" or "Save", and one `*.json` file per `use-case`. 
3. Still have a `CommonTranslation.json`, but have `*.json` everywhere, for every use-case, screen, or component per basis. I.e. a `Button.comp.translations.json` for translations for a Button.
```

File: react/README.md [hunk 3]
```diff
@@ -9,6 +9,277 @@ For migration between style guides, please see [MIGRATION.md](MIGRATION.md).
...
+
+### Component
+
+A component is everything that's not a [use-case](#use-case) or [screen](#screen). The only important thing to remember is to give a component only responsibility for one thing. In other words, follow the Single-responsibility principle.
+
+---
+
+## File naming
+
+Each folder, including use-cases, screens, views, and elements, can, but must not, contain these types of files:
```
Comment (ericlp) [inline react/README.md]: Shouldn't it be Components not Elements?

File: react/README.md [hunk 4]
```diff
@@ -9,6 +9,277 @@ For migration between style guides, please see [MIGRATION.md](MIGRATION.md).
...
+-   ShowUsers, UserDetail: Two screens that are a part of the [use-case](#use-case) Users.
+
+
+### Component
+
+A component is everything that's not a [use-case](#use-case) or [screen](#screen). The only important thing to remember is to give a component only responsibility for one thing. In other words, follow the Single-responsibility principle.
+
+---
+
+## File naming
```
Comment (ericlp) [inline react/README.md:49]: You don't mention that files can be named `Element.comp.jsx` but include them in the example below.

File: react/README.md (hunk 2 above)
Comment (ericlp) [reply react/README.md]: Yeah, that is a clearer

File: react/README.md (hunk 3 above)
Comment (Portals) [reply react/README.md]:
```suggestion
Each folder, including use-cases, screens, and components, can, but must not, contain these types of files:
```

File: react/README.md (hunk 4 above)
Comment (Portals) [reply react/README.md:49]: I changed it to this and committed:

```md
## File naming

Each folder, including use-cases, screens, and components, can, but must not, contain these types of files:

* `index.js`, should contain the export of the `*.jsx` file. If there's a container, however, it should export that instead. index.js is used to you only have to import the folder, not a specific file inside the folder.

* `MyUseCase.jsx` / `MyScreen.screen.jsx` / `MyComponent.comp.jsx`: the main element.

* `Element.translations.json`/`Element.(screen/comp).translations.json`: has the translation data for the Element.
```

Comment (Portals) [reply react/README.md:49]: image

**Review by ericlp — APPROVED** [review]

**Pull request merged**